redwood-treehouse/src/hostMain/kotlin/app/cash/redwood/treehouse/TreehouseApp.kt

```diff
@@ -16,18 +16,19 @@
 package app.cash.redwood.treehouse
 
 import app.cash.redwood.protocol.Diff
+import app.cash.redwood.protocol.Event
 import app.cash.redwood.protocol.EventSink
 import app.cash.redwood.protocol.widget.DiffConsumingNode
 import app.cash.redwood.protocol.widget.ProtocolBridge
 import app.cash.redwood.widget.Widget
 import app.cash.zipline.Zipline
-import app.cash.zipline.ZiplineService
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.job
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.yield
```

Comment:
Oooh what a great shame, being caught yielding! Because you demand to know what I was doing with that naughty device...

There’s a crash when doing a code swap in EmojiSearch. I think it might be just the 60hz frame pulse being enqueued and executing too late, though I haven’t really locked that down. I wanted to try yield to see if it was just a race in how different coroutines were sequenced during tear down. That wasn‘t it.

The real fix is for me to do the [shutdown thing](https://github.com/cashapp/zipline/issues/694) I’ve been sitting on for forever in Zipline. My first PR works but I’m not happy with how it works specifically around enqueued jobs. Time to do some hard design on what the requirements there should be.

Reply:
Hah I immediately did a search for its usage and was relieved when it came up empty.